src/Test/QuickCheck/Gen.purs

```diff
@@ -103,33 +105,33 @@ frequency x xs = let
     pick n d [] = d
     pick n d ((Tuple k x) : xs) = if n <= k then x else pick (n - k) d xs
   in do
-    n <- chooseInt 1 total
+    n <- choose 0 total
```

Comment:
What do you think of this?

`chooseInt` doesn't quite seem right, as the weights are not integers, likewise starting at 1 seems too large a number, as the combined weights might not add up to 1 even. Unless I've misunderstood.